cmake/Modules/LibtorrentMacros.cmake

```diff
@@ -1,80 +1,126 @@
-# Various helper function and macros for building libtorrent
-
+# Various helper functions and macros for building libtorrent
 include(FeatureSummary)
 
-# macro for issuing option() and add_feature_info() in a single call.
+# function for issuing option() and add_feature_info() in a single call.
 # Synopsis:
-# feature_option(<option_and_feature_name> <option_and_feature_description> <default_option_value>)
-macro(feature_option _name _description _default)
-	option(${_name} "${_description}" ${_default})
-	add_feature_info(${_name} ${_name} "${_description}")
+# feature_option(<option/feature name> <description> <default value>)
+function(feature_option _name _description _default)
+    string(CONCAT _desc "${_description} (default: ${_default})")
+    option("${_name}" "${_desc}" "${_default}")
+    add_feature_info("${_name}" "${_name}" "${_desc}")
+endfunction()
+
+# Set common variables and create some interface-only library targets
+# that project targets can link to, either directly or transitively,
+# to consume common compile options/definitions
+macro(libtorrent_common_config)
```

Comment:
this is set up in a macro, not to be re-used, but to allow it to live in a separate file, is that right?

Reply:
@arvidn 

Yes. We do something similar in qBittorrent.